server/jetstream_cluster.go

```diff
@@ -7914,27 +7925,36 @@ func (mset *stream) processClusteredInboundMsg(subject, reply string, hdr, msg [
 		mset.clseq = lseq + mset.clfs
 	}
 
-	// Check if we have an interest policy and discard new with max msgs or bytes.
+	// Check if we have an interest or working queue retention and discard new with max msgs or bytes.
 	// We need to deny here otherwise it could succeed on some peers and not others
 	// depending on consumer ack state. So we deny here, if we allow that means we know
 	// it would succeed on every peer.
-	if interestPolicy && discard == DiscardNew && (maxMsgs > 0 || maxBytes > 0) {
+	if interestPolicy && discard == DiscardNew && (maxMsgs > 0 || maxBytes > 0 || (maxMsgsPer > 0 && discardNewPerSubject)) {
 		// Track inflight.
 		if mset.inflight == nil {
 			mset.inflight = make(map[uint64]uint64)
 		}
+
 		if stype == FileStorage {
 			mset.inflight[mset.clseq] = fileStoreMsgSize(subject, hdr, msg)
 		} else {
 			mset.inflight[mset.clseq] = memStoreMsgSize(subject, hdr, msg)
 		}
 
+		if mset.inflightSubjects == nil {
+			mset.inflightSubjects = make(map[string]uint64)
+		}
+
```

Comment:
Remove nl

Reply:
Can not remove newline as go fmt puts it back

Reply:
Really? It does not for me..